### PR TITLE
fix: throw error if target not found with name

### LINKED ISSUE #332: Do not use a fallback target if provided target name was not found
In such a case, a warning `Target with the name 'targetName' was not found in sasjsconfig.json.` should be printed only. CLI shouldn't use a fallback target in such a case.

### LINKED ISSUE #495: when running `sasjs add` and target already exists, provide an option to replace it
Would be nice if the prompt framework would let you choose whether to replace a target (and if so, it would take defaults from it)

image

### LINKED ISSUE #500: Error / Warning on console for compile
there is one thing that keeps coming up every time - there is an ERROR in the console when doing a compile
it should probably be a NOTE

image

The "WARN" could also be removed.
- If a target is not found, the processing should stop.
- If the target is not provided (as in this case) then it should default to the first target with a NOTE

>It's dangerous if I intend to deploy to say, -t dev but I type -t ddev and it defaults to -t prod

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

Closes #332
Closes #500 
Closes #495

## Intent

- If target is specified in any `sasjs <command>`, and not found it should stop and do not fall back to first.
- if target name is not specifed in any `sasjs <command`, then fallback message should be _INFO_ not _WARN_
- While `sasjs add target`, if specified `targetName` is found, instead of giving error, show option to update this `target`

## Implementation

Error throw in `findTargetInConfiguration`, if targetName is provided and not found and changed logger warning -> info
`getCommonFields` in `addTarget.ts` is changed and also returns `existingTarget`, added new user input `getAndValidateUpdateExisting`
`add-target.spec.ts` is refactored and added more tests to it.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).

mocked: 
image

server:
image

- [ ] All CI checks are green.
- [x] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

File: src/commands/add/internal/input.ts [hunk 1]
```diff
@@ -117,6 +123,44 @@ async function getAndValidateTargetName(
...
+      ]
+    ).catch(() => {
+      throw new Error(errorMessage)
+    })
+
+    if (choice === null || choice === undefined || Number.isNaN(choice)) {
+      throw new Error(errorMessage)
+    }
+
+    console.log('Choice: ', choice)
```
Comment (YuryShkoda) [inline src/commands/add/internal/input.ts:157]: do we need this `console.log` ?

File: src/commands/add/internal/input.ts [hunk 2]
```diff
@@ -117,6 +123,44 @@ async function getAndValidateTargetName(
...
+      'Please choose either option 1 or 2.',
+      [
+        { title: '1. Update this target', value: 1 },
+        { title: '2. Go back and create a new target', value: 2 }
+      ]
+    ).catch(() => {
+      throw new Error(errorMessage)
+    })
+
+    if (choice === null || choice === undefined || Number.isNaN(choice)) {
```
Comment (YuryShkoda) [inline src/commands/add/internal/input.ts:153]: maybe we can just check `if (!choice)...`

File: src/commands/add/spec/add-target.spec.ts [hunk 3]
```diff
@@ -15,6 +15,15 @@ import { TargetJson } from '../../../types/configuration'
 import { TargetScope } from '../../../types/targetScope'
 import { createTestMinimalApp, removeTestApp } from '../../../utils/test'
 
+interface CommonFields {
```
Comment (YuryShkoda) [inline src/commands/add/spec/add-target.spec.ts:18]: it makes sense to put this interface in `types` folder

File: src/utils/config.ts [hunk 4]
```diff
@@ -94,7 +99,7 @@ export async function findTargetInConfiguration(
       : undefined
 
     if (fallBackTargetJson) {
-      process.logger?.warn(
+      process.logger?.info(
         `Target ${targetName || ''} was not found. Falling back to target ${
```
Comment (YuryShkoda) [inline src/utils/config.ts:103]: `targetName` will always be an empty string here

File: src/utils/config.ts [hunk 5]
```diff
@@ -117,7 +122,7 @@ export async function findTargetInConfiguration(
     : undefined
 
   if (fallBackTargetJson) {
-    process.logger?.warn(
+    process.logger?.info(
```
Comment (YuryShkoda) [inline src/utils/config.ts:125]: `targetName` will always be an empty string here

File: src/commands/add/internal/input.ts (hunk 1 above)
Comment (saadjutt01) [reply src/commands/add/internal/input.ts:157]: We are using same for getting `Scope`
image

Should we change it to logger.Info?

Comment (saadjutt01) [reply src/commands/add/internal/input.ts:157]: removed from both places.

File: src/commands/add/internal/input.ts (hunk 2 above)
Comment (saadjutt01) [reply src/commands/add/internal/input.ts:153]: fixed

File: src/utils/config.ts (hunk 4 above)
Comment (saadjutt01) [reply src/utils/config.ts:103]: got it, fixed.

**Review by YuryShkoda — CHANGES_REQUESTED** [review]

File: src/commands/add/spec/add-target.spec.ts (hunk 3 above)
Comment (saadjutt01) [reply src/commands/add/spec/add-target.spec.ts]: fixed

File: src/utils/config.ts (hunk 5 above)
Comment (saadjutt01) [reply src/utils/config.ts:123]: got it, fixed

**Review by YuryShkoda — APPROVED** [review]

**Pull request merged**